### DISCUSSION TITLE
Feature Request: NULL for non-set env vars from getenv

### DISCUSSION THREAD

**Discussion by tmontes** [duckdb/duckdb#14641] [Ideas]
PREAMBLE:

* Thanks for your work in building duckdb: it's a wonderful tool that's a joy to use. :-)

CONTEXT:

* I've been building specifically purposed Docker containers around the `duckdb` CLI, leveraging the `getenv` function to parametrize their behaviour (example: country and/or year selection).
* Such containers are easy to test and compose into wider scoped data analysis pipelines.

STATUS:

* As of 1.1.2, the `getenv` function returns an empty string if the given VARIABLE is not set.

REQUEST:

* `getenv` could (should?) return NULL if the VARIABLE is not set.
   (only returning the empty string if the VARIABLE is set to that in the environment)

MOTIVATION:

* More consistent behaviour where NULL would represent undefined/unset.
  (much like in many SQL scenarios)

IMPLEMENTATION:

* If acceptable, I might take a look at producing a PR (with guidance!)

**Comment by Tishj** [comment]
Sounds sane, perhaps `default` should be an optional parameter, set to NULL by default

Similar to: https://docs.python.org/3/library/os.html#os.getenv

**Reply by tmontes** [reply]
...nothing that a `COALESCE(getenv('NOT_DEFINED'), 'default-value')` couldn't handle, but why not? :-)
